projects/npm-tools/packages/npm-scripts/test/utils/expandGlobs.js

```diff
@@ -220,4 +220,22 @@ describe('expandGlobs()', () => {
 			/Redundant ignore patterns/
 		);
 	});
+
+	it('can match files in a different directory', () => {
+		const randomTempDir = fs.mkdtempSync(
+			path.join(os.tmpdir(), 'scripts-basedir-')
+		);
+
+		const tempFilePath = path.posix.join(randomTempDir, 'test.js');
```

Comment:
Anyone know why it is necessary to use `posix` here? Windows was failing earlier with this error, but adding posix seemed to fix it. However, looking at the output, I'm not totally sure why it only mattered for that last directory path.

```
    - Expected  - 1
    + Received  + 1

      Array [
    -   "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\scripts-basedir-0RbmjO\\test.js",
    +   "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\scripts-basedir-0RbmjO/test.js",
      ]
```

Reply:
Actually, I don't think its right to use `path.posix` because now I am just making the expected value to use both `\\`  and `/`.

cc @wincent @jbalsas any ideas?

Reply:
It doesn't make sense to use `path.posix` there. In fact, it almost never makes sense unless you are reading posix paths from a configuration or something like that (in which case, I usually resort to string functions `split` and `join` passing `/` directly to make it explicit that we are not dealing with paths, but with a multi-platform logical representation).

Nevertheless, Windows APIs are POSIX, so you can use posix paths for Windows too (though it may lead to problems because some older APIs in libc are not POSIX and you never know for sure what APIs the applications are using).

Said that, I don't know why using `path.posix` there fixes the test. My bet is that the underlying functions you are using for expanding globs are mixing POSIX and Windows paths. For example: they can be taking the paths in the format you pass them but then completing paths with POSIX, or something like that...

Reply:
> My bet is that the underlying functions you are using for expanding globs are mixing POSIX and Windows paths.

That's a new behavior. We have always expressed globs using POSIX format, and when expanded them we have always returned them in expanded form [in the same format](https://github.com/liferay/liferay-frontend-projects/blob/19e03ab7adbaf20d7ad23b3f6235bb18ab9430a2/projects/npm-tools/packages/npm-scripts/src/utils/expandGlobs.js#L130).

The difference is that in this PR @bryceosterhaus has taught `expandGlobs()` to take `baseDir` param. So, previously, we would always operate relative to the current working directory, which meant that given a glob like `**/*.js` we would expand that to things like `a/b.js`, `a/c.js` etc regardless of platform and everything was rosy. But with `baseDir` in effect, on Windows the glob now expands to `SOME_WINDOWS_PATH\a/b.js` etc.

So, we can do any of these:

- Call it harmless and leave the test like it is (with the `path.posix` call), possibly with a comment explaining why:

  ```
  // Note that globs are always expressed in (and expanded to) POSIX form.
  const tempFilePath = path.posix.join(randomTempDir, 'test.js');
  ```
- Back out the change that added `baseDir` support and instead `process.chdir()` in `collectSassFile()`; might need to do a `path.join()` in there if we definitely need absolute paths (I have no idea if we do).
- Change `expandGlobs()` to stop using `path.posix.join()` (but note that it's been like that [for two years now](https://github.com/liferay/liferay-frontend-projects/commit/5e76049d782b85614c274ffd95d064a1ba6c8332), and that commit flags at least one more thing we'd need to change in order to keep things working on Windows if we do that).

My preference is to do the minimal thing (ie. the first option) just to minimize the risk of regressions.